tests/integ/test_open_telemetry.py

```diff
@@ -0,0 +1,375 @@
+
+    # test register with @udaf
+    @udaf(
+        name="sum_udaf",
+        session=session,
+        input_types=[IntegerType()],
+        return_type=IntegerType(),
+        replace=True,
+    )
+    class PythonSumUDAF:
+        def __init__(self) -> None:
+            self._sum = 0
+
+        @property
+        def aggregate_state(self):
+            return self._sum
+
+        def accumulate(self, input_value):
+            self._sum += input_value
+
+        def merge(self, other_sum):
+            self._sum += other_sum
+
+        def finish(self):
+            return self._sum
+
+    spans = spans_to_dict(dict_exporter.get_finished_spans())
+    assert "register" in spans
+    span = spans["register"]
+    assert (
+        os.path.basename(span.attributes["code.filepath"]) == "test_open_telemetry.py"
+    )
+    assert span.attributes["snow.executable.name"] == "sum_udaf"
+    assert span.attributes["snow.executable.handler"] == "PythonSumUDAF"
+    dict_exporter.clear()
+
+
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="UDTF not supported in Local Testing",
+)
+def test_register_udtf_from_file(session, dict_exporter):
+    test_file = os.path.normpath(
+        os.path.join(
+            os.path.dirname(__file__),
+            "../resources",
+            "test_udtf_dir",
+            "test_udtf_file.py",
+        )
+    )
+
+    schema = StructType(
+        [
+            StructField("int_", IntegerType()),
+            StructField("float_", FloatType()),
+            StructField("bool_", BooleanType()),
+            StructField("decimal_", DecimalType(10, 2)),
+            StructField("str_", StringType()),
+            StructField("bytes_", BinaryType()),
+            StructField("bytearray_", BinaryType()),
+        ]
+    )
+
+    lineno = inspect.currentframe().f_lineno + 1
+    session.udtf.register_from_file(
+        test_file,
+        "MyUDTFWithTypeHints",
+        name="MyUDTFWithTypeHints_from_file",
+        output_schema=schema,
+        input_types=[
+            IntegerType(),
+            FloatType(),
+            BooleanType(),
+            DecimalType(10, 2),
+            StringType(),
+            BinaryType(),
+            BinaryType(),
+        ],
+        immutable=True,
+        replace=True,
+    )
+    spans = spans_to_dict(dict_exporter.get_finished_spans())
+    assert "register_from_file" in spans
+    span = spans["register_from_file"]
+    assert (
+        os.path.basename(span.attributes["code.filepath"]) == "test_open_telemetry.py"
+    )
+    assert span.attributes["code.lineno"] == lineno
+    assert span.attributes["snow.executable.name"] == "MyUDTFWithTypeHints_from_file"
+    assert span.attributes["snow.executable.handler"] == "MyUDTFWithTypeHints"
+    dict_exporter.clear()
+
+
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="UDTF not supported in Local Testing",
+)
```

Comment:
The top level mark for this file should already be skipping on this condition so no need to add it here.

Reply:
got it